clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2276,6 +2337,141 @@ private Node leader(PartitionInfo partitionInfo) {
         return partitionInfo.leader();
     }
 
+    @Override
+    public void describeTopics(
+        TopicCollection topics,
+        DescribeTopicsOptions options,
+        AdminResultsSubscriber<DescribeTopicPartitionsResult> subscriber) {
+        if (topics instanceof TopicIdCollection) {
+            subscriber.onError(
+                new IllegalArgumentException("Currently the describeTopics subscription mode does not support topic IDs.")
+            );
+            return;
+        }
+        if (!(topics instanceof TopicNameCollection)) {
+            subscriber.onError(
+                new IllegalArgumentException("The TopicCollection: " + topics + " provided did not match any supported classes for describeTopics.")
+            );
+            return;
+        }
+
+        TreeSet<String> topicNames = new TreeSet<>();
+        ((TopicNameCollection) topics).topicNames().forEach(topicName -> {
+            if (topicNameIsUnrepresentable(topicName)) {
+                subscriber.onNext(DescribeTopicPartitionsResult.onError(topicName, new InvalidTopicException("The given topic name '" +
+                    topicName + "' cannot be represented in a request.")));
+            } else {
+                topicNames.add(topicName);
+            }
+        });
+
+        RecurringCall call = new RecurringCall(
+            "DescribeTopics-Recurring",
+            calcDeadlineMs(time.milliseconds(), options.timeoutMs()),
+            runnable
+        ) {
+
+            Map<String, TopicRequest> pendingTopics = new TreeMap<>();
+            Iterator<String> pendingTopicIterator = topicNames.iterator();
+
+            String partiallyFinishedTopicName = "";
+            int partiallyFinishedTopicNextPartitionId = -1;
```

Comment:
Could we replace this with a DescribeTopicPartitionsRequestData.Cursor instead of the two separate variables? It makes its purpose a bit more clear and we don't have to be worried about them getting out of sync.

```suggestion
            DescribeTopicPartitionsRequestData.Cursor requestCursor = null;
```